README.md

```diff
@@ -1,8 +1,152 @@
-#Cabin
+# Cabin
 
 This repo contains Cabin, a tool to facilitate the versioning of database tables.
 
 
 ## Dev Environment Setup
+Assuming linux machine, macOS may need to install some extras such as wget (through brew or from source).
 
-To get your Cabin dev environment up and running, run `bash mysql/init.sh`. This will run the officialy MySQL Docker image and create a Cabin database and user on it.
+1. Set up the repo:
+```
+git clone github.com/seeqbio/cabin.git
+cd cabin
+make venv
+source venv/bin/activate
+```
+
+2. Set MySQL: 
+To get a dev environment up and running for mysql, either (1) run `bash mysql/init.sh`. This will run the officialy MySQL Docker image and create a Cabin database and user on it. **Or** if you have mysql already set up (2) add the name of the database you wish to use, the user, and password to the environmental variables in `cabin/settings.py`. Then run `cabin init` to create a `cabin_system` table.
+
+3. Quick Start:
+Below are commands to run to import the example data provided. See usage examples and documentation for more details.
+
+```
+(venv) > bin/cabin list
+(venv) > bin/cabin status
+(venv) > bin/cabin import --all --dry-run
+(venv) > bin/cabin import --all
+(venv) > bin/cabin status
+(venv) > bin/cabin shell
+```
+
+## Background
+This project came out of a startup where "cabin" was used for data collection and maintence. We sought to share some aspects of this tool we developed with the community. Carving out the tool from our code base resulted in some messiness that we tried to take care of but bare with us if artifacts are found here and there; please point them out and we would gladly sort out any confusion. We are aware of several areas/features that could use our attention, but again, it would be great to priortize based on needs or confusions that arise.
+
+## Usage examples
+*The following is an example, with detailed output, of how a dataset would be imported and updated*
+
+**See which datasets are available to work with, import all.** Note that:
+
+* the `import` command may specify a dataset directly or use `--all` to import all possible datasets that are not already imported
+* given the one example, `import --all` is effectively the same as specifying `import StormDetailsTable` because all dependancies of the table are verified either way
+* `import StormDetailsFile` however would only have triggered the download of the csv to the downloads directory
+* `--dry-run` allows a preview of which datasets will be affected
+
+```
+(venv) $ bin/cabin list
+StormDetailsFile
+StormDetailsOfficial
+StormDetailsTable
+(venv) cabin $ bin/cabin import --all
```

Comment:
what's the first `cabin`?

Reply:
working directory after user has cloned and `cd cabin`. is it better to assume that and keep the prompt as `$`?